In [1]:
import numpy as np

In [28]:
# Enter the data
mat = np.asmatrix([(0,2,1,2,-1),(2,6,4,5,-1)])

In [29]:
mat

matrix([[ 0,  2,  1,  2, -1],
        [ 2,  6,  4,  5, -1]])

In [32]:
# Compute the covariance matrix
cov = np.cov(mat)

In [33]:
cov

array([[1.7 , 3.55],
       [3.55, 7.7 ]])

In [35]:
# Compute the eigenvectors and eigenvalues
eigvals, eigvecs = np.linalg.eig(cov)

In [37]:
eigvals

array([0.05215104, 9.34784896])

The 2nd eigenvalue is larger, so this is the *first* principal component.

In [38]:
eigvecs

array([[-0.90704462, -0.42103452],
       [ 0.42103452, -0.90704462]])

The 2nd column is the 2nd eigenvector: [ -0.42103452, -0.90704462]. This corresponds to the 2nd eigenvalue. So the first component of this vector is -0.42103452.